cg/services/post_processing/pacbio/metrics_parser/utils.py

```diff
@@ -42,37 +48,42 @@ def parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) -> B
 
 
 @handle_pac_bio_parsing_errors
-def parse_dataset_metrics(report_dir: Path) -> SmrtlinkDatasetsMetrics:
-    file_name = PacBioDirsAndFiles.SMRTLINK_DATASETS_REPORT
-    report: Path = get_file_in_directory(directory=report_dir, file_name=file_name)
-    parsed_json: dict = ReadFile.read_file[FileFormat.JSON](report)
+def parse_dataset_metrics(metrics_files: list[Path]) -> SmrtlinkDatasetsMetrics:
+    report_file: Path = get_report_file_from_pattern(
```

Comment:
If we add another function:

```Python
get_data_model_for_pattern()
```

Then we can handle all files /models in one function. What do you think?

Reply:
Good idea, implementing to see how it looks